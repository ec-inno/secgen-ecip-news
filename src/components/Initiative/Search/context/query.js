import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
