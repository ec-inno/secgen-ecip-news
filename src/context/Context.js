import React from 'react';

const initialState = {
  hideOverlay: true,
};

export const SET_LANGUAGE_OVERLAY_VISIBILITY =
  'SET_LANGUAGE_OVERLAY_VISIBILITY';

const reducer = (state, action) => {
  switch (action.type) {
    case SET_LANGUAGE_OVERLAY_VISIBILITY: {
      return Object.assign({}, state, {
        hideOverlay: action.hideOverlay,
      });
    }

    default: {
      return initialState;
    }
  }
};

const Context = React.createContext();

const ContextProviderWithReducer = ({ children }) => {
  const [store, dispatch] = React.useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ store, dispatch }}>{children}</Context.Provider>
  );
};

export { ContextProviderWithReducer, Context as default };
