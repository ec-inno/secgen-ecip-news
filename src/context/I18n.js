import React, { useContext } from 'react';

const I18nContext = React.createContext();

const useI18nContext = () => useContext(I18nContext);

export { useI18nContext, I18nContext as default };
