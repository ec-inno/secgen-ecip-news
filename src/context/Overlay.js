import React, { useContext } from 'react';

const Context = React.createContext();

const useOverlayContext = () => useContext(Context);

export { useOverlayContext, Context as default };
