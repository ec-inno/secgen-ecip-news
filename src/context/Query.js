import React, { useContext } from 'react';

import { defaultLangKey } from '../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'unsetFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

const useQueryContext = () => useContext(Context);

export { queryInit, queryReducer, Context as default, useQueryContext };
