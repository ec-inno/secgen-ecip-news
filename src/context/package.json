{
  "name": "@eci/context",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@ecl/eu-preset-website": "2.8.0",
    "react": "16.9.0"
  }
}
