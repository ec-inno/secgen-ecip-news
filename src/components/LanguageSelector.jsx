import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import Context, { SET_LANGUAGE_OVERLAY_VISIBILITY } from '../Context';

import Icon from './Icon';

const LanguageSelector = ({ href, name, code, opensOverlay }) => {
  const { dispatch } = useContext(Context);

  const openOverlay = e => {
    e.preventDefault();

    dispatch({
      type: SET_LANGUAGE_OVERLAY_VISIBILITY,
      hideOverlay: false,
    });
  };

  const nothing = e => e.preventDefault();

  return (
    <div className="ecl-site-header__selector">
      <a
        onClick={opensOverlay ? openOverlay : nothing}
        className="ecl-link ecl-link--standalone"
        href={href}
        data-ecl-language-selector
      >
        {name}
        <span className="ecl-site-header__language-icon">
          <Icon shape="general--language" size="m" />
          <span className="ecl-site-header__language-code">{code}</span>
        </span>
      </a>
    </div>
  );
};

LanguageSelector.propTypes = {
  opensOverlay: PropTypes.bool,
  href: PropTypes.string,
  name: PropTypes.string,
  code: PropTypes.string,
};

export default LanguageSelector;
