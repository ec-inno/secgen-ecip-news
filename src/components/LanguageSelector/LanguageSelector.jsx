import React from 'react';
import PropTypes from 'prop-types';

import { useOverlayContext } from '../../context/Overlay';

import Icon from '../Icon';

const LanguageSelector = ({ href, name, code }) => {
  const { setOverlayIsHidden } = useOverlayContext();

  return (
    <div className="ecl-site-header__selector">
      <a
        onClick={e => {
          e.preventDefault();
          setOverlayIsHidden(false);
        }}
        className="ecl-link ecl-link--standalone"
        href={href}
        data-ecl-language-selector
      >
        {name}
        <span className="ecl-site-header__language-icon">
          <Icon shape="general--language" size="m" />
          <span className="ecl-site-header__language-code">{code}</span>
        </span>
      </a>
    </div>
  );
};

LanguageSelector.propTypes = {
  href: PropTypes.string,
  name: PropTypes.string,
  code: PropTypes.string,
};

LanguageSelector.defaultProps = {
  href: '',
  name: '',
  code: '',
};

export default LanguageSelector;
