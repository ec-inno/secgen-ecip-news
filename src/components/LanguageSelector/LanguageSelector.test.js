import React from 'react';
import renderer from 'react-test-renderer';

import * as OverlayContext from '../../context/Overlay';
import LanguageSelector from './LanguageSelector';

describe('LanguageSelector', () => {
  it('renders correctly', () => {
    jest.spyOn(OverlayContext, 'useOverlayContext').mockImplementation(() => ({
      overlayIsHidden: true,
      setOverlayIsHidden: () => true,
    }));

    const props = {
      href: '#',
      name: 'English',
      code: 'en',
    };

    const tree = renderer.create(<LanguageSelector {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
