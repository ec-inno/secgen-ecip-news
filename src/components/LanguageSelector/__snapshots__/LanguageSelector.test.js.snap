// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector renders correctly 1`] = `
<div
  className="ecl-site-header__selector"
>
  <a
    className="ecl-link ecl-link--standalone"
    data-ecl-language-selector={true}
    href="#"
    onClick={[Function]}
  >
    English
    <span
      className="ecl-site-header__language-icon"
    >
      <svg
        aria-hidden="true"
        className="ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--language"
        />
      </svg>
      <span
        className="ecl-site-header__language-code"
      >
        en
      </span>
    </span>
  </a>
</div>
`;
